src/main/java/picard/sam/util/ReadNameParser.java

```diff
@@ -26,15 +30,15 @@
      */
     public static final String DEFAULT_READ_NAME_REGEX = "<optimized capture of last three ':' separated fields as numeric values>".intern();
 
-    private final int[] tmpLocationFields = new int[3]; // for optimization of addLocationInformation
+    private transient final int[] tmpLocationFields = new int[3]; // for optimization of addLocationInformation
 
     private String readNameRegex = null;
 
-    private Pattern readNamePattern;
+    private transient Pattern readNamePattern;
```

Comment:
Is this not serializable or just an optimization to reduce serialization size?  If it's the second case I'd make it non-transient.